### PR TITLE
(#140) feat: fairing OnLaunch

### PR DESCRIPTION
**Pull request by dannypsnl**
@shana0440 after you reviewed I would add the document, so don't directly merge it.

### CODE REVIEW

File: fairing/fairing.go [hunk 1]
```diff
@@ -6,10 +6,21 @@ import (
...
 )
 
+// Meta is created for fairing OnLaunch
+type Meta struct {
+	// Port is the serve port of rocket server
+	Port string
+}
+
 // Interface specify the method that fairing could implement
 type Interface interface {
```
[the remark this answers is not in the document]
Comment (dannypsnl) [reply fairing/fairing.go]: Actually, I'm thinking do we still need sub-pkg `fairing`? When I'm working on this PR, I found I need the information about **rocket**, and can't be removed, `Meta` duplicate with `rocket.Rocket`.

So I think maybe they should be the same package level, what do you think?

[the remark this answers is not in the document]
Comment (shana0440) [reply fairing/fairing.go]: If you want to remove `fairing` package, I'm don't have the objection.

[the remark this answers is not in the document]
Comment (dannypsnl) [reply fairing/fairing.go]: Now this interface located at `rocket` named `FairingInterface`. Do you think we need to private it?

[the remark this answers is not in the document]
Comment (shana0440) [reply fairing/fairing.go]: yes, I think user don't need to know this interface.

[the remark this answers is not in the document]
Comment (dannypsnl) [reply fairing/fairing.go]: Could you try the difference between their GoDoc? I thought a private interface with a public method `Attach` might be confused?

**Pull request merged**